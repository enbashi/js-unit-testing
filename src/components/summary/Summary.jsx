import React, { Component, PropTypes } from 'react';
import _ from 'lodash';
import {
  formatProductLabel
} from '../../redux/modules/summary/summaryHelpers';
import {
  formatAmount
} from '../../util/helpers';

class Summary extends Component {
  componentDidMount(){
    this.props.loadProducts();
  }

  render() {
    const {
      updateProduct,
      updateQuantity,
      products,
      total,
      selectedProductId,
      selectedQuantity,
    } = this.props;

    if(_.isEmpty(products)){
      return <h1 className="loading">Loading......</h1>;
    }

    return (
      <div className="main-content">

          <form>
              <div className="form-title-row">
                  <h1>Welcome to my store</h1>
              </div>
              <div className="form-row">
                  <label>
                      <span>Product</span>
                      <select defaultValue={selectedProductId} onChange={(e) => updateProduct(e.target.value)}>
                        <option>Select</option>
                        {
                          products.map((product, index) => {
                            return (<option key={index} value={product.id}>{formatProductLabel(product)}</option>);
                          })
                        }
                      </select>
                  </label>
              </div>
              <div className="form-row">
                  <label>
                      <span>Quantity</span>
                      <select defaultValue={selectedQuantity} onChange={(e) => updateQuantity(parseInt(e.target.value, 10))}>
                          <option>Select</option>
                          <option>1</option>
                          <option>2</option>
                          <option>3</option>
                          <option>4</option>
                          <option>5</option>
                      </select>
                  </label>
              </div>
              {
                total &&
                <div className="form-row total">
                    <label>
                        <span>Total</span>
                    </label>
                    <span>{formatAmount(total)}</span>
                </div>
              }
          </form>
      </div>
    );
  }
}

Summary.propTypes = {
  products: PropTypes.array,
  selectedProductId:  PropTypes.string,
  selectedQuantity: PropTypes.number,
  total: PropTypes.number,
  loadProducts: PropTypes.func.isRequired,
  updateProduct: PropTypes.func.isRequired,
  updateQuantity: PropTypes.func.isRequired
};

export default Summary;
