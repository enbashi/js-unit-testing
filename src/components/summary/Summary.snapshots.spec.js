import React from 'react';
import Summary from './Summary';
import renderer from 'react-test-renderer';

describe('Summary Snapshots', () => {
  it('renders Loading when no products are available yet', () => {
    const component = renderer.create(
      <Summary
        products={[]}
        loadProducts={() => {}}
        updateProduct={() => {}}
        updateQuantity={() => {}}
      />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders products', () => {
    const component = renderer.create(
      <Summary
        products={[
          {
            id: 111,
            name: 'X1 Cound DVR',
            price: 10.00
          },
          {
            id: 222,
            name: 'DTA Additional Outlet',
            price: 3.99
          }
        ]}
        loadProducts={() => {}}
        updateProduct={() => {}}
        updateQuantity={() => {}}
      />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
