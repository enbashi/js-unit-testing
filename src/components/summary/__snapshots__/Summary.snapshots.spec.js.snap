exports[`Summary Snapshots renders Loading when no products are available yet 1`] = `
<h1
  className="loading">
  Loading...
</h1>
`;

exports[`Summary Snapshots renders products 1`] = `
<div
  className="main-content">
  <form>
    <div
      className="form-title-row">
      <h1>
        Welcome to my store
      </h1>
    </div>
    <div
      className="form-row products">
      <label>
        <span>
          Product
        </span>
        <select
          defaultValue={undefined}
          onChange={[Function]}>
          <option>
            Select
          </option>
          <option
            value={111}>
            X1 Cound DVR - $ 10.00
          </option>
          <option
            value={222}>
            DTA Additional Outlet - $ 3.99
          </option>
        </select>
      </label>
    </div>
    <div
      className="form-row quantity">
      <label>
        <span>
          Quantity
        </span>
        <select
          defaultValue={undefined}
          onChange={[Function]}>
          <option>
            Select
          </option>
          <option>
            1
          </option>
          <option>
            2
          </option>
          <option>
            3
          </option>
          <option>
            4
          </option>
          <option>
            5
          </option>
        </select>
      </label>
    </div>
  </form>
</div>
`;
